charmcraft/commands/store/__init__.py

```diff
@@ -1396,26 +1355,50 @@ def run(self, parsed_args):
             resource_type = ResourceType.file
             logger.debug("Uploading resource directly from file %s", resource_filepath)
         elif parsed_args.image:
+            image_digest = parsed_args.image
+            credentials = store.get_oci_registry_credentials(
+                parsed_args.charm_name, parsed_args.resource_name
+            )
+            image_name = credentials.image_name.split("/", 1)[1]
```

Comment:
store.get_oci_registry_credentials' .image_name always returned a / (or validated), right?

Reply:
from this operation it looks like the image_name is not really the image_name, can you add a code comment on why this is?

Reply:
The credentials.image_name is a standard OCI registry construct, example: `'registry.jujucharms.com/charm/45kk8smbsqp2f90ssgtygsp8atuzup4biyn2e/redis-image'`.

The "image name" from the credentials is this full image name used by OCI Registries. But the "image name" that we handled internally is without the server part.

Maybe we could improve this by changing `credentials` to not hold "image_name" internally, but "full_image_name" or something...

Reply:
I suspect this is obvious as it is fresh in your head, which is why I thought a code comment for a variable/attribute that had the same name but with different comments for future self would be nice (or a different name as suggested)

Reply:
Using a different variable name if the nature of the content changes would be a good idea IMHO.